scripts/remove-idle-users.py

```diff
@@ -37,43 +37,49 @@ def kick_idlers(homeserver, homeserver_domain, room_id, token, since, user_prefi
         # Ignore users that do not start with the user_prefix
         if not user_id.startswith(user_prefix):
             continue
-        res = requests.post(
-            homeserver + "/_matrix/client/r0/rooms/" + urllib.quote(room_id) + "/kick?access_token=" + token,
+        res = requests.put(
+            homeserver + "/_matrix/client/r0/rooms/" +
+            urllib.quote(room_id) + "/state/m.room.member/" +
+            urllib.quote(user_id) + "?access_token=" + token,
             data = json.dumps({
                 "reason": reason,
-                "user_id": user_id
+                "membership": "leave"
             })
         )
         if res.status_code >= 400:
-            failure_responses.append({ "user_id": user_id, "response_json": res.json()})
+            failure = { "user_id": user_id }
+            try:
+                failure["response_json"] = res.json()
+            except Exception as e:
+                print("Could not get JSON body from failure response: %s" % e)
+            failure_responses.append()
         else:
             count += 1
-    print("Kicked %s users in total" % count)
+    print("Kicked %s/%s users in total (%s failed requests)" % (count, len(user_ids), len(failure_responses)))
```

Comment:
My earlier comment was explicit for a reason: `len(user_ids)` will include ones you want to filter out according to `prefix`. It's probably fine since you're giving all the information required to work out the values, but it'll be confusing to see things like:

> Kicked 5/10 users in total (2 failed requests)

and wonder what happened to the remaining 3.